packages/flutter/lib/src/services/system_navigator.dart

```diff
@@ -35,4 +35,33 @@ class SystemNavigator {
   static Future<void> pop({bool? animated}) async {
     await SystemChannels.platform.invokeMethod<void>('SystemNavigator.pop', animated);
   }
+
+  /// Notifies the platform for a route information change.
+  ///
+  /// On Web, creates a new browser history entry and update URL with the route
+  /// information stores in the [args].
+  static void routeInformationUpdated(Object args) {
+    // TODO(chunhtai): Invokes routeInformationUpdated once the engine side is done.
+    final Map<dynamic, dynamic> argsMap = args as Map<dynamic, dynamic>;
+    SystemChannels.navigation.invokeMethod<void>(
```

Comment:
I have to reuse the old method until the engine side method is in place otherwise it will throw an exception.

Reply:
Instead of duplication the code, forward the call to routeUpdated?